File: python/leap/leap.py
def is_leap_year(year):
    return True if is_divisible_by(year, 4) and (not is_divisible_by(year, 100) or is_divisible_by(year, 400)) else False

def is_divisible_by(number, divisor):
    return (number % divisor) == 0

File: go/leap/leap.go
package leap

// IsLeapYear returns true if the year would have an extra day
func IsLeapYear(year int) bool {
	return isDivisbleBy(year, 4) && (!isDivisbleBy(year, 100) || isDivisbleBy(year, 400))
}

func isDivisbleBy(number int, divisor int) bool {
	return number%divisor == 0
}
